go/pools/resource_pool.go

```diff
@@ -104,9 +104,12 @@ func (rp *ResourcePool) TryGet() (resource Resource, err error) {
 
 func (rp *ResourcePool) get(ctx context.Context, wait bool) (resource Resource, err error) {
 	// If ctx has already expired, avoid racing with rp's resource channel.
-	if ctx.Err() != nil {
+	select {
+	case <-ctx.Done():
```

Comment:
https://github.com/golang/net/blob/master/context/context.go#L91

ok, but then the code need to check ctx.Err() and then return nil, ErrTimeout, right ?